packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2372,35 +2364,30 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
     markParentNeedsLayout();
   }
 
-  /// Set [_relayoutBoundary] to null throughout this render object's subtree,
+  /// Set [_isRelayoutBoundary] to null throughout this render object's subtree,
   /// stopping at relayout boundaries.
   // This is a static method to reduce closure allocation with visitChildren.
   static void _cleanChildRelayoutBoundary(RenderObject child) {
-    if (child._relayoutBoundary != child) {
+    if (child._isRelayoutBoundary != true) {
       child.visitChildren(_cleanChildRelayoutBoundary);
-      child._relayoutBoundary = null;
+      child._isRelayoutBoundary = null;
     }
   }
 
   // This is a static method to reduce closure allocation with visitChildren.
   static void _propagateRelayoutBoundaryToChild(RenderObject child) {
-    if (child._relayoutBoundary == child) {
+    if (child._isRelayoutBoundary ?? false) {
       return;
     }
-    final RenderObject? parentRelayoutBoundary = child.parent?._relayoutBoundary;
-    assert(parentRelayoutBoundary != null);
-    assert(parentRelayoutBoundary != child._relayoutBoundary);
-    child._setRelayoutBoundary(parentRelayoutBoundary!);
+    assert(child.parent?._isRelayoutBoundary != null);
+    child._setIsRelayoutBoundary(false);
   }
 
-  /// Set [_relayoutBoundary] to [value] throughout this render object's
-  /// subtree, including this render object but stopping at relayout boundaries
-  /// thereafter.
-  void _setRelayoutBoundary(RenderObject value) {
-    assert(value != _relayoutBoundary);
-    // This may temporarily break the _relayoutBoundary invariant at children;
-    // the visitChildren restores the invariant.
-    _relayoutBoundary = value;
+  /// Set [_isRelayoutBoundary] to [value] on this render object,
+  /// then to false throughout the rest of its subtree,
+  /// stopping at relayout boundaries.
+  void _setIsRelayoutBoundary(bool value) {
```

Comment:
The name of this method makes it seem as if you can replace it with `_isRelayoutBoundary = value`. Maybe indicate in the name that this includes recursing down the tree?

Reply:
Also, some of this walking is now unnecessary as you've indicated in your comment, right? Maybe leave a TODO somewhere here that you can remove in your next PR to avoid confusion for anyone reading this code before that PR lands.

Reply:
Actually, I am wondering why we want to do this in two separate PRs instead of including it all in this one? If the next step for whatever reason doesn't work out leaving the code in this state is kinda awkward.